src/mscorlib/shared/System/Text/StringBuilder.cs

```diff
@@ -1645,6 +1645,34 @@ public bool Equals(StringBuilder sb)
             }
         }
 
+        /// <summary>
+        /// Determines if the contents of this builder are equal to the contents of ReadOnlySpan<char>.
+        /// </summary>
+        /// <param name="value">The ReadOnlySpan<char>.</param>
+        public bool Equals(ReadOnlySpan<char> value)
+        {
+            if (value.Length != Length)
+                return false;
+
+            StringBuilder thisChunk = this;
+            int offset = 0;
+            int chunk_length = -1;
+            
+            
+            while (thisChunk != null)
```

Comment:
We know `this` isn't null, so this loop could be a do-while instead of a while.

Reply:
Nit: `thisChunk` is also a bit of a misnomer, as it only refers to `this` for the first iteration.